### PR TITLE
IA-2213 add warning for user with no perm

### PR DESCRIPTION
**Pull request by quang-le**
Add a warning when creating a user ith no permissions  through the web interface that the created user will have access to the mobile app but not to the features of the web app

Related JIRA tickets : IA-2213

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [X] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- Switch UsersDialog to TS
- Move logic related to the conversion of `initialData`to the format needed by the component to `useInitialData` hook
- Pass `isLoading`to Table to force refresh when updating a user
- use `isLoading` to force refresh `UsersDialog` when creating a user

The main problem was that The modal uses the old `ConfirmCancelModal` component and that converting it to the new component would have entailed rewriting the cypress tests, which would have inflated the scope of the ticket and delayed it. The current solution is a bit hacky but should make the feature available while doing a proper rewrite

## How to test

- Go to users 
- try to create a user with no permissions
- try to add permissions to that user
- try to remove all permissions from a user

## Print screen / video

https://github.com/BLSQ/iaso/assets/38907762/66e51845-05cb-4e7e-901a-ea2f36a13752

## Notes
Ticket for full modal upgrade https://bluesquare.atlassian.net/browse/IA-2238

### CODE REVIEW

**Comment by quang-le** [comment]
@cypress

File: hat/assets/js/apps/Iaso/domains/app/translations/en.json [hunk 1]
```diff
@@ -991,13 +991,15 @@
...
     "iaso.users.label.noTypeAssigned": "No org unit type assigned",
+    "iaso.users.labels.createUserWithoutPerm": "Save user with no permissions?",
     "iaso.users.newPassword": "New password",
     "iaso.users.ouChildrenCheckbox": "Users with access to children org unit",
     "iaso.users.ouParentCheckbox": "Users with access to parent org unit",
     "iaso.users.password": "Password",
     "iaso.users.permissions": "Permissions",
     "iaso.users.selectedOrgUnits": "Org units selected",
     "iaso.users.update": "Update user",
+    "iaso.users.warningModalMessage": "You are about to save a user with no permissions. This user will have access to the mobile application but not to the features of the web interface.",
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/app/translations/en.json:1015]: without permissions ?

File: hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx [hunk 2]
```diff
@@ -36,100 +43,49 @@ const useStyles = makeStyles(theme => ({
     },
 }));
 
-const UserDialogComponent = ({
+type Props = {
+    titleMessage: IntlMessage;
+    renderTrigger: any;
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx]: ReactElement ?

File: hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx [hunk 3]
```diff
@@ -36,100 +43,49 @@ const useStyles = makeStyles(theme => ({
...
+    initialData?: InitialUserData;
+    saveProfile: any;
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx]: Promise<any>  ?

File: hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx [hunk 4]
```diff
@@ -36,100 +43,49 @@ const useStyles = makeStyles(theme => ({
...
+    const saveUser = useCallback(() => {
+        const currentUser: any = {};
         Object.keys(user).forEach(key => {
             currentUser[key] = user[key].value;
         });
 
         saveProfile(currentUser, {
             onSuccess: () => {
-                closeDialog();
+                // closeDialog();
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx]: you can remove this ?

File: hat/assets/js/apps/Iaso/domains/users/components/WarningModal/WarningModal.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+}) => {
+    const { formatMessage } = useSafeIntl();
+    return (
+        <ConfirmCancelModal
+            open={open}
+            id="user-WarningModal"
+            dataTestId="user-WarningModal"
+            titleMessage={formatMessage(MESSAGES.createUserWithoutPerm)}
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            onClose={() => {}}
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/WarningModal/WarningModal.tsx]: `onClose={() => undefined}` ?

File: hat/assets/js/apps/Iaso/domains/users/components/WarningModal/WarningModal.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            closeDialog={closeDialog}
+            onConfirm={onConfirm}
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            onCancel={() => {}}
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/users/components/WarningModal/WarningModal.tsx]: same

**Review by beygorghor — CHANGES_REQUESTED** [review]
Small remarks + see screen shot 
Screenshot 2023-06-14 at 12 53 21

File: hat/assets/js/apps/Iaso/domains/app/translations/en.json (hunk 1 above)
Comment (quang-le) [reply hat/assets/js/apps/Iaso/domains/app/translations/en.json:1015]: Yeah, I wasn't sure about that one

File: hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx (hunk 3 above)
Comment (quang-le) [reply hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx]: MutateFunction<Profile, any>

File: hat/assets/js/apps/Iaso/domains/app/translations/en.json (hunk 1 above)
Comment (quang-le) [reply hat/assets/js/apps/Iaso/domains/app/translations/en.json:1015]: I think it's OK as is

**Review by beygorghor — APPROVED** [review]

**Pull request merged**